great_expectations/util.py

```diff
@@ -1295,14 +1296,64 @@ def is_nan(value: Any) -> bool:
     Returns:
         The results of the test
     """
-    import numpy as np
-
     try:
-        return np.isnan(value)
+        return isnan(value)
     except TypeError:
         return True
 
 
+def _is_nan_or_none(scalar: Any) -> bool:
+    """Return True if scalar is NaN or None"""
+    try:
+        result = isnan(scalar)
+    except TypeError:
+        result = scalar == None
+    return result
+
+
+def compare_two_lists_or_items_that_might_have_nan(first: Any, second: Any) -> bool:
+    """Did you know that `float("nan") != float("nan")`?
+
+    Only 1 level of nesting is supported by this func
+    """
+    result = False
+    if first == second:
+        result = True
```

Comment:
Could we just return here? And at the end of each conditional branch? We don't need to evaluate the rest of the function of `result` gets set right?